### PR TITLE
logging: Add logging

### PR DESCRIPTION
**Pull request by Dawnkai**
Add loggers and replace print statements with logging.

Closes https://github.com/chauffer/super/issues/18.

### CODE REVIEW

File: super/cogs/youtube.py [hunk 1]
```diff
@@ -68,9 +71,8 @@ async def leave(self, ctx, server, message):
             if self.S[ctx.message.guild.id].is_connected:
                 return await self.S[ctx.message.guild.id].disconnect()
         except:
-            return await ctx.message.channel.send(
-                ":thinking:\n " + traceback.format_exc()
-            )
+            logger.error(traceback.format_exc())
```
Comment (chauffer) [inline super/cogs/youtube.py:74]: I want the traceback to be sent to the channel, I don't want to read the logs to find out what happened, if you want to log, do that on top

File: super/__init__.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+            'level': 'DEBUG',
+            'formatter': 'standard',
+            'class': 'logging.StreamHandler',
+            'stream': 'ext://sys.stdout'
+        },
+        'file': {
+            'level': 'DEBUG',
+            'formatter': 'standard',
+            'class': 'logging.handlers.RotatingFileHandler',
+            'filename': SUPER_LOG_FILE_NAME,
```
Comment (chauffer) [inline super/__init__.py:25]: let's not do files, stdout is enough

File: super/cogs/youtube.py (hunk 1 above)
Comment (Dawnkai) [reply super/cogs/youtube.py]: Restored.

File: super/__init__.py (hunk 2 above)
Comment (Dawnkai) [reply super/__init__.py]: Removed.

File: super/cogs/astro.py [hunk 3]
```diff
@@ -41,6 +45,7 @@ async def _astro(self, ctx, owo=False):
             message = ctx.message.content.split(" ")
 
             if len(message) >= 2:
+                logger.debug("astro_sunsign: Fetching", message=message[1])
```
Comment (chauffer) [inline super/cogs/astro.py:48]: I like the idea of using this format, but two events should not be the same.

I would do `cogs/astro/_astro: Fetching`

File: super/cogs/astro.py [hunk 4]
```diff
@@ -28,6 +31,7 @@ def __init__(self, bot):
         self.api = "http://horoscope-api.herokuapp.com/horoscope/{when}/{sunsign}"
 
     async def _get_sunsign(self, sunsign, when):
+        logger.debug("astro_sunsign: Fetching", sunsign=sunsign, when=when)
```
Comment (chauffer) [inline super/cogs/astro.py:34]: `cogs/astro/_get_sunsign: ...`

File: super/settings.py [hunk 5]
```diff
@@ -25,3 +25,5 @@
 SUPER_MAX_YOUTUBE_LENGTH = int(os.getenv("SUPER_MAX_YOUTUBE_LENGTH", 3600))
 
 SUPER_YOUTUBE_TIMEOUT = int(os.getenv('SUPER_YOUTUBE_TIMEOUT', 600))
+
+SUPER_LOG_MODE = os.getenv("SUPER_LOG_MODE", "PRODUCTION")
```
Comment (chauffer) [inline super/settings.py:29]: let's make
`SUPER_DEBUG = os.getenv("SUPER_DEBUG") is not None`
instead. this is simpler as setting `SUPER_DEBUG` to anything will make it `True` and might be used elsewhere other than logging as well.

File: super/cogs/astro.py (hunk 3 above)
Comment (Dawnkai) [reply super/cogs/astro.py]: Changed to `<folder>/<module>/<function>`

File: super/cogs/astro.py (hunk 4 above)
Comment (Dawnkai) [reply super/cogs/astro.py]: Changed

File: super/settings.py (hunk 5 above)
Comment (Dawnkai) [reply super/settings.py]: Replaced.

File: super/cogs/f1.py [hunk 6]
```diff
@@ -23,7 +26,8 @@ async def calendar(self):
         except:
             return self._calendar
 
-    async def get_events(self, num=10, page=0, more=False, weekend=False):
+    async def get_events(self, num, page=0, more=False):
```
Comment (chauffer) [inline super/cogs/f1.py:29]: hey put the weeknd back pls

File: super/cogs/youtube.py [hunk 7]
```diff
@@ -89,8 +95,10 @@ async def skip(self, ctx, server, message):
         try:
             server.voice_client.stop()
             await ctx.message.channel.send("skipping")
+            logger.info("cogs/youtube/skip: Skipped song", song=server.playing.url)
```
Comment (chauffer) [inline super/cogs/youtube.py:98]: i would log server id here too

**Review by chauffer — APPROVED** [review]

**Pull request merged**